update: multiplatform resources page restructured (#227)
JetBrains/kotlin-multiplatform-dev-docs: topics/compose/compose-resources-usage.md

[//]: # (title: Using multiplatform resources in your app)

<toc-settings structure-elements="chapter" depth="3"/>

> When you've [set up the resources for your project](compose-resources-setup.md),

Reviewer: I feel like now this intro is only concerned with the Res class generation and doesn't mention other important sections on the page, consider adding another paragraph, for example:

```
When you've set up the resources for your project, first generate the special Res class, which provides access to resources. 

To generate the Res class and all the resource accessors, build the project or re-import the project in the IDE.

<a paragraph about different resources usage and accessing external resources>

## Class generation settings﻿

You can also customize the generated class to suit your needs using the Gradle settings.

In the compose.resources {} block, ...
```

Author: True, true. I adjusted the beginning mostly like you suggested.

<tabs>
<tab title= "From composable code">

```kotlin
var bytes by remember {
    mutableStateOf(ByteArray(0))
}
LaunchedEffect(Unit) {
    bytes = Res.readBytes("files/myDir/someFile.bin")
}
Text(bytes.decodeToString())
```

</tab>
<tab title= "From non-composable code">

```kotlin
coroutineScope.launch {
    val bytes = Res.readBytes("files/myDir/someFile.bin")
}
```

</tab>
</tabs>

#### Convert byte arrays into images {label="EAP"}

If the file you are reading is a bitmap (JPEG, PNG, BMP, WEBP) or an XML vector image, you can use the following functions
to convert them into `ImageBitmap` or `ImageVector` objects suitable for the `Image()` composable.

Access the raw files as shown in the [Raw files](#raw-files) section, then pass the result to a composable:

```kotlin
// bytes = Res.readBytes("files/example.png")
Image(bytes.decodeToImageBitmap(), null)

// bytes = Res.readBytes("files/example.xml")
Image(bytes.decodeToImageVector(LocalDensity.current), null)
```

On every platform except Android, you can also turn an SVG file into a `Painter` object:

```kotlin
// bytes = Res.readBytes("files/example.svg")
Image(bytes.decodeToSvgPainter(LocalDensity.current), null)
```

> ### Using Java resources with Compose Multiplatform

Reviewer: ```suggestion
### Using Java resources
```

Reviewer: Some headers are to big for the right-side menu, I think this one can be safely shortened

```

An example of passing a mapped resource to a composable:

```kotlin
Image(painterResource(Res.allDrawableResources["compose_multiplatform"]!!), null)
```

## Accessing resources from external libraries

If you want to use multiplatform resources in other libraries included in your project, you can pass the platform-specific
path to the file to them.
To get the platform-specific path, call the `Res.getUri()` function with the project path to the resource:

```kotlin
val uri = Res.getUri("files/my_video.mp4")
```

Now that the `uri` variable contains the absolute path to the file, any external library can use that path to access
the file in a manner that suits it.

### Compose Multiplatform resources as Android assets {label="EAP"}

Starting with Compose Multiplatform %composeEapVersion%, all multiplatform resources are packed into Android assets.
This enables Android Studio to generate previews for Compose Multiplatform composables in Android source sets.

> Android Studio previews are available only for composables in an Android source set.
> They also require one of the latest versions of AGP: 8.5.2, 8.6.0-rc01, or 8.7.0-alpha04.
>
{type="warning"}

> Multiplatform resources as Android assets also allow directly accessing them on Android from WebViews and media player components,

Reviewer: ```suggestion
Using Multiplatform resources as Android assets also means direct access from WebViews and media player components on Android
```

> since resources can be reached by a simple path, for example `Res.getUri(“files/index.html”)`

Reviewer: ```suggestion
since resources can be reached by a simple path, for example `Res.getUri("files/index.html")`.
```
